### ISSUE TITLE
Config.php and configuration.php always created on docker-compose up

### ISSUE THREAD

**Issue by peremp** [directus/directus-docker#8]
Any time I do a docker-compose up, the _configuration.php_ file is generated with the content of _configuration.stub_.

So, even if I create a volume for the file _configuration.php_, the modified file is replaced on each _docker-compose up_ with the content of _configuration.stub_

Any suggestion on how to solve this? Should I modify the configuration.stub file?

**Comment by WellingGuzman** [comment]
I don't know if I'm understanding correctly, but `configuration.php` file is created based on `configuration.stub`, and every time you run `docker-compose up` the `configuration.php` is recreated.

**Comment by peremp** [comment]
Exactly, and that's the problem, because I have created some hooks on configuration.php... and every time I do a composer-up those changes are lost.

**Comment by WellingGuzman** [comment]
@peremp that's how `docker-compose up` should work, it re-create the container? I'm not an expert in the area, but it seems that's how it should work.

Maybe you are trying to find a different command, what are you trying to accomplish when you run `docker-compose up` after you have created the container?

**Comment by Froelund** [comment]
We could make a check, to see if the `configuration.php` already exists [before doing directus install](https://github.com/directus/directus-docker/blob/master/directus-base-layer/usr/local/bin/run-directus#L60). I guess it would make sense, when you map those files into the container from file-system. Although I might rather want to map-overwrite the stub if I were you. That way you can still manage the database using environment vars.

Have you considered making an optional `configuration.extra.stub` for this purpose @WellingGuzman ? A file that would be included in the final configuration for some custom configs.

**Comment by peremp** [comment]
@WellingGuzman The problem is that the install script is launched after the volumes are synchronized, so the _config.php_ & _configuration.php_ are overwritten. And since the configuration file is used to define hooks/filters, this is an issue.

@Froelund I think that your solution could work.

Anyway, a co-worker is also working on a solution so we will keep you up to date.

**Comment by Froelund** [comment]
Sounds good @peremp.

**Comment by mawrkus** [comment]
I've found a simple workaround for our use case: I create .stub files for each environment (dev, staging, ...) and a Dockerfile to copy them into the container (depending on the target environment).
This way we can control/tweak the content of the config files generated.

**Comment by WellingGuzman** [comment]
hey @mawrkus, would you care to show us how this works? :)

**Comment by mawrkus** [comment]
Sure! This is what my staging Dockerfile looks like: 

```yaml
FROM getdirectus/directus:6.3

COPY ./src/config/config.staging.stub /var/www/html/api/core/Directus/Util/Installation/stubs/config.stub
COPY ./src/config/configuration.stub /var/www/html/api/core/Directus/Util/Installation/stubs/configuration.stub
COPY ./src/hooks/*.php /var/www/html/api/core/Directus/Custom/Hooks/
```

And my docker-compose.override.yml (for local development):

```yaml
version: '3'

services:
  directus:
    image: getdirectus/directus:6.3
    volumes:
      - ./src/config/config.dev.stub:/var/www/html/api/core/Directus/Util/Installation/stubs/config.stub
      - ./src/config/configuration.stub:/var/www/html/api/core/Directus/Util/Installation/stubs/configuration.stub
      - ./src/hooks/:/var/www/html/api/core/Directus/Custom/Hooks/

...
```

**Comment by WellingGuzman** [comment]
Thanks @mawrkus, I'd try to squeeze this into the docker repo, maybe this can be helpful to someone else. (can we use it?)

@peremp does this PR https://github.com/directus/directus-docker/pull/9 and/or the above workaround solve your issue?

**Comment by peremp** [comment]
Yes. Thank you all ;)

**Comment by mawrkus** [comment]
@WellingGuzman sure, feel free to use it!

**Comment by WellingGuzman** [comment]
Thanks I will close this as @peremp solved the problem with PR#9.

I will try to implement @mawrkus solution into the repository, if someone else would like to try to incorporate into the repo we would love to. @mawrkus ? :)
